addons/stock/report/report_stock_forecasted.py

```diff
@@ -159,13 +159,15 @@ def _reconcile_out_with_ins(lines, out, ins, demand, product_rounding, only_matc
             product_template_ids, product_variant_ids, wh_location_ids
         )
         outs = self.env['stock.move'].search(out_domain, order='reservation_date, priority desc, date, id')
-        reserved_outs = self.env['stock.move'].search(
-            out_domain + [('state', 'in', ('partially_available', 'assigned'))],
-            order='priority desc, date, id')
         outs_per_product = defaultdict(list)
         reserved_outs_per_product = defaultdict(list)
+        outs_reservation = {}
         for out in outs:
             outs_per_product[out.product_id.id].append(out)
+            outs_reservation[out.id] = out._get_orig_reserved_availability()
+        reserved_out_ids = outs.filtered(lambda m: not float_is_zero(outs_reservation[m.id], precision_rounding=m.product_uom.rounding)).ids
+        reserved_outs = self.env['stock.move'].search([('id', 'in', reserved_out_ids)], order='priority desc, date, id')
```

Comment:
a browse and a sorted would be faster no?

Reply:
Unless I misunderstood when I checked it, `sorted()` doesn't allow to have one sort key being reversed (here `priority desc`) mixed with regular sort keys (here `date` and `id`).